lib/membrane_dashboard/charts/helpers.ex

```diff
@@ -55,43 +73,126 @@ defmodule Membrane.Dashboard.Charts.Helpers do
   end
 
   @doc """
-  Gets rows of TimescaleDB's measurements table and `interval` as list of timestamps.
+  Gets rows of TimescaleDB's `measurements` table and `interval` as list of timestamps.
 
   Returns list of tuples `{path, data}`, where `path` is pipeline element path and data is a list with
   values (one value for every timestamp in `interval`).
   """
-  @spec to_series([[term()] | binary()] | nil, [float()]) :: [{String.t(), [non_neg_integer()]}]
+  @spec to_series(postgrex_result_rows_t(), [float()]) :: [{String.t(), [non_neg_integer()]}]
   def to_series(rows, interval) do
     rows
-    |> Enum.group_by(fn [_time, path, _size] -> path end, fn [time, _path, size] ->
-      {time, size}
+    |> rows_to_data_by_paths()
+    |> data_by_paths_to_series(interval)
+  end
+
+  @doc """
+  Gets rows of TimescaleDB's `measurements` table, `interval` as list of timestamps, `mode` which is either `:full` or `:update` and two arguments needed when mode is set to `:update`:
+  - old data - 2D list contatining metric data before update
+  - paths - list of all paths that will be present in the new data
+
+  Returns list of tuples `{path, data}`, where `path` is pipeline element path and data is a list with values (one value for every timestamp in `interval`). 
+  Data is altered in the way that every non-nil value is a number of processed metric events from the beginning of live update.
+  """
+  @spec to_series(postgrex_result_rows_t(), [float()], atom(), [[non_neg_integer()]], [
```

Comment:
To be honest this type spec is wild, by that I mean it looks almost like something auto generated and does not tell much about return value structure until you spend some amount of time studying it. If you don't want to create a separate type spec use this syntax `measurement :: float(), label :: atom()` ect.